### PR TITLE
Pdf delivery report

### LINKED ISSUE #1452: Delivery report pdf export
Skulle det vara möjligt att kunna ladda ned även denna fil som en PDF?

Leveransrapport Clinical Genomics
https://scout-stage.scilifelab.se/cust002/F0032094/delivery-report

### PR DESCRIPTION
**Pull request by northwestwitch**
fix #1452

Delivery report can be exported from the side panel on the case page. I've introduced two icons which are links to respectively: html page and pdf download:

image

**How to test**:
1. Bash install this branch on scout stage. Don't forget to restart the server with the command:
`supervisorctl restart scout-stage`
1. Go to a case which has one or more analyses with delivery report and test if the report is downloadable.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
Writing some tests now!

**Comment by northwestwitch** [comment]
Ready for review!

File: scout/server/blueprints/cases/views.py [hunk 1]
```diff
@@ -777,10 +778,23 @@ def delivery_report(institute_id, case_name):
...
+    format = request.args.get('format','html')
+    if format == 'pdf':
+        try: # file could not be available
+            html_file = open(delivery_report, 'r')
+            source_code = html_file.read()
+            # remove image, since it is problematic to render it in the PDF version
+            source_code=re.sub('<img class=.*?alt="SWEDAC logo">','',source_code, flags=re.DOTALL)
+            return render_pdf(HTML(string=source_code), download_filename=case_obj['display_name']+'_'+datetime.datetime.now().strftime("%Y-%m-%d")+'_scout_delivery.pdf')
+        except Exception as ex:
+            flash('An error occurred while download {} -- {}'.format(delivery_report, ex), 'warning')
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py]: "during download of" or "while downloading". 😄

File: tests/server/blueprints/cases/test_cases_views.py [hunk 2]
```diff
@@ -414,3 +414,54 @@ def test_status(app, institute_obj, case_obj, user_obj):
...
+            {'$set' : {'delivery_report': delivery_report_path} }
+        )
+
+        # WHEN accessing the delivery report page
+        resp = client.get(url_for('cases.delivery_report',
+                                  institute_id=institute_obj['internal_id'],
+                                  case_name=case_obj['display_name']))
+
+        # THEN the endpoint should return the delivery report HTML page
+        assert 'Leveransrapport Clinical Genomics' in str(resp.data)
```
Comment (dnil) [inline tests/server/blueprints/cases/test_cases_views.py]: :+1:

File: tests/server/blueprints/cases/test_cases_views.py [hunk 3]
```diff
@@ -414,3 +414,54 @@ def test_status(app, institute_obj, case_obj, user_obj):
...
+        # WHEN accessing the delivery report page with the format=pdf param
+        resp = client.get(url_for('cases.delivery_report',
+                                  institute_id=institute_obj['internal_id'],
+                                  case_name=case_obj['display_name'],
+                                  format='pdf'))
+
+        # a successful response should be returned
+        assert resp.status_code == 200
+        # and it should contain a pdf file, not HTML code
+        assert resp.mimetype == 'application/pdf'
```
Comment (dnil) [inline tests/server/blueprints/cases/test_cases_views.py]: :+1:

**Comment by dnil** [comment]
Looks good; I'll check it out on stage as well..

File: scout/server/blueprints/cases/views.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/views.py]: Haha yes! :)

**Comment by dnil** [comment]
In particular, the html icon is just fine, but the pdf one just opens another html view..

**Comment by northwestwitch** [comment]
> In particular, the html icon is just fine, but the pdf one just opens another html view..

Which case, all of them?

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html [hunk 4]
```diff
@@ -64,15 +64,33 @@
...
-          <span class="menu-collapsed">Delivery</span>
-      </a>
+      <div href="#" class="bg-dark list-group-item text-white">
+        <div class="d-flex align-items-center justify-content-start">
+            <div><span class="menu-collapsed ml-2">Delivery</span></div>
+            <div>
+              <a href="{{ url_for('cases.delivery_report', institute_id=institute._id,
+                              case_name=case.display_name) }}" target="_blank">
+                              <i class="fa fa-link"></i></a>
+              <a href="{{ url_for('cases.delivery_report', institute_id=institute._id,
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html]: No target here, which should be fine I suppose..

**Comment by dnil** [comment]
Yeah, but now all seems fine and dandy with https://scout-stage.scilifelab.se/cust002/F0015031. I wonder if I just had something partially cached?!

**Comment by northwestwitch** [comment]
> Yeah, but now all seems fine and dandy with https://scout-stage.scilifelab.se/cust002/F0015031. I wonder if I just had something partially cached?!

Maybe?

**Comment by northwestwitch** [comment]
> Ive a tested a couple of cases, and all that can render the html seem able to render the pdf now. And request favicon. 😆

Sweet! I don't actually see the favicon message since a couple of days so I start to feel worried.. :D

Anyway, can we merge?

**Pull request merged**